phoenix-core/src/it/java/org/apache/phoenix/end2end/DateTimeIT.java

```diff
@@ -1939,6 +1941,99 @@ public void testDateFormatTimeZone()throws Exception {
+        stmt.execute("CREATE TABLE " + tableName + " ( id INTEGER not null PRIMARY KEY," +
+                " date DATE, time TIME, timestamp TIMESTAMP)");
+
+        stmt.execute("UPSERT INTO " + tableName + " VALUES(1, TO_DATE('" + date1
+                + "'), TO_TIME('" + date1 + "'), TO_TIMESTAMP('" + date1 + "'))");
+
+        stmt.execute("UPSERT INTO " + tableName + " VALUES(2, TO_DATE('" + date2
+                + "'), TO_TIME('" + date2 + "'), TO_TIMESTAMP('" + date2 + "'))");
+
+        String f = " GMT', 'yyyy-MM-dd HH:mm:ss.SSS z', 'UTC";
+        stmt.execute("UPSERT INTO " + tableName + " VALUES(4, TO_DATE('" + date2 + f
+                + "'), TO_TIME('" + date2 + f + "'), TO_TIMESTAMP('" + date2 + f + "'))");
+
+        PreparedStatement pstmt = conn.prepareStatement("UPSERT INTO " + tableName + " values (?, ?, ?, ?)");
+        pstmt.setInt(1, 3);
+        Timestamp t =  new Timestamp(DateUtil.parseDate(date3).getTime());
+        pstmt.setDate(2, new Date(t.getTime()));
+        pstmt.setTime(3, new Time(t.getTime()));
+        pstmt.setTimestamp(4, t);
+        pstmt.execute();
+        conn.commit();
+
+        ResultSet rs = stmt.executeQuery("SELECT * FROM " + tableName + " ORDER BY id");
+
+        int assertions = 0;
+        for (int i = 0; i < dates.size(); i++){
+            assertTrue(rs.next());
+            String actualDate = dates.get(i);
+            Timestamp parseTimestamp = DateUtil.parseTimestamp(actualDate);
+
+            assertEquals(i + 1, rs.getInt(1));
+            Date expectedDate = DateUtil.getSQLDateInISOChronology(parseTimestamp);
+            assertEquals(expectedDate, rs.getDate(2));
+            assertEquals(rs.getDate(2), rs.getObject(2));
+            assertEquals(actualDate, rs.getString(2));
+
+            Time expectedTime = DateUtil.getSQLTimeInISOChronology(parseTimestamp);
+            assertEquals(new Timestamp(expectedTime.getTime()), new Timestamp(rs.getTime(3).getTime()));
+            assertEquals(expectedTime, rs.getTime(3));
+            assertEquals(rs.getTime(3), rs.getObject(3));
+            assertEquals(actualDate, rs.getString(3));
+
+            Timestamp expectedTimestamp = DateUtil.getSQLTimestampInISOChronology(parseTimestamp);
+            assertEquals(expectedTimestamp, rs.getTimestamp(4));
+            assertEquals(rs.getTimestamp(4), rs.getObject(4));
+            assertEquals(actualDate, rs.getString(4));
+            assertions++;
+        }
+
+        String query = "SELECT year(timestamp), month(timestamp), dayofmonth(timestamp),"
+                + " hour(timestamp), minute(timestamp), second(timestamp) FROM "  + tableName
+                + " ORDER BY id";
+        rs = stmt.executeQuery(query);
+
+        for (int i = 0; i < dates.size(); i++){
+            assertTrue(rs.next());
+            for (int j = 0; j < 6; j++){
+                int expected = Integer.parseInt(dateLists.get(i).get(j));
```

Comment:
nit: IMO, would be more readable to add `List<String> dateList = dateList.get(i)` outside the inner-loop and make this `dateList.get(j)`.